player_darwin.c

```diff
@@ -0,0 +1,143 @@
+//
+// Created by Christopher Cooper on 7/12/17.
+//
+
+#include <sys/param.h>
+#include "player_darwin.h"
+
+#pragma mark - utility functions
+
+void MakeBufferSilent (AudioBufferList * ioData) {
+
+  for(UInt32 i=0; i<ioData->mNumberBuffers;i++) {
+    memset(ioData->mBuffers[i].mData, 0, ioData->mBuffers[i].mDataByteSize);
+  }
+
+}
+
+// the following two functions are here because Go can't work with variable length arrays - it always sees them as an array
+// with a single element
+
+void RenderFloatBufferData(UInt32 bufferIndex, UInt32 frame, Float32 f, AudioBufferList *ioData) {
+
+  Float32 *data = (Float32*)ioData->mBuffers[bufferIndex].mData;
+  data[frame] = f;
+  //data[frame +1] = f;
```

Comment:
This PR isn't meant to be final, commented out things will be removed in the final version